test/CMakeLists.txt

```
@@ -243,5 +243,28 @@ celeritas_setup_tests(SERIAL PREFIX field)
 celeritas_cudaoptional_test(field/RungeKutta)
 celeritas_cudaoptional_test(field/FieldDriver)
 
+if(CELERITAS_USE_VecGeom)
+  celeritas_setup_tests(SERIAL PREFIX field
+    LINK_LIBRARIES CeleritasPropagatorTest VecGeom::vecgeom
+  )
+  celeritas_add_test(field/FieldPropagator.test.cc GPU)
+
+  if(CELERITAS_USE_CUDA)
+    add_library(CeleritasPropagatorTest field/FieldPropagator.test.cu)
+    target_link_libraries(CeleritasPropagatorTest
+      PRIVATE 
+      celeritas
+      VecGeom::vecgeomcuda
+      VecGeom::vecgeomcuda_static
+    )
+    set_target_properties(CeleritasPropagatorTest PROPERTIES
+      LINKER_LANGUAGE CUDA
+      CUDA_SEPARABLE_COMPILATION ON
+      POSITION_INDEPENDENT_CODE ON
+    )
```

I added a CMake macro at some point to hide the extra flags (and so we can change them all in one place if needed). I  think this should work:

```suggestion
    target_link_libraries(CeleritasPropagatorTest
      PRIVATE celeritas
    )
    celeritas_link_vecgeom_cuda(CeleritasPropagatorTest)
```

---

Thanks for a simplification!  Modified accordingly.